lib/feedback_hook.rb

```diff
@@ -0,0 +1,104 @@
+class CppFeedbackHook < Mumukit::Hook
+  def run!(request, results)
+    content = request.content
+    test_results = results.test_results[0]
+
+    CppExplainer.new.explain(content, test_results) if test_results.is_a? String
```

Comment:
I think we should return an empty string instead of `nil` if the results are structured. Could you double check that, please?